code/datums/traits/negative.dm

```diff
@@ -579,3 +579,49 @@
 	lose_text = "<span class='notice'>Your mind finally feels calm.</span>"
 	medical_record_text = "Patient's mind is in a vulnerable state, and cannot recover from traumatic events."
 	hardcore_value = 9
+
+/datum/quirk/allergic
+	name = "Extreme Medicine Allergy"
+	desc = "Ever since you were a kid you always were allergic to certain chemicals..."
+	value = -2
+	gain_text = "<span class='danger'>You feel your immune system shift.</span>"
+	lose_text = "<span class='notice'>You fell your immune system phase back into perfect shape.</span>"
+	medical_record_text = "Patient's immune system responds violently to certain chemicals."
+	hardcore_value = 3
+	var/list/allergies = list()
+
+/datum/quirk/allergic/on_spawn()
+	var/list/chem_list = subtypesof(/datum/reagent/medicine)- /datum/reagent/medicine/c2 - /datum/reagent/medicine/epinephrine
+	for(var/i in 0 to 10)
+		var/chem = pick(chem_list)
+		chem_list -= chem
+		allergies += chem
+
+/datum/quirk/allergic/post_add()
+	var/display = ""
+	for(var/C in allergies)
+		var/datum/reagent/chemical = C
+		display += initial(chemical.name) + ", "
+	name = "Extreme " + display +"Allergies"
+	medical_record_text = "Patient's immune system responds violently to [display]"
+	quirk_holder.add_memory("You are extremely allergic to : [display]")
+	to_chat(quirk_holder, "<span class='boldnotice'>You are allergic to [display]make sure not to consume any of it!</span>")
+
+/datum/quirk/allergic/on_process()
+	. = ..()
+	if(!iscarbon(quirk_holder))
+		return
+	var/mob/living/carbon/carbon_quirk_holder = quirk_holder
+	for(var/M in allergies)
+		var/datum/reagent/med = M
+		var/datum/reagent/instantiated_med = carbon_quirk_holder.reagents.has_reagent(med)
+		if(instantiated_med)
+			//Just halts the progression, I'd suggest you run to medbay asap to get it fixed
+			if(carbon_quirk_holder.reagents.has_reagent(/datum/reagent/medicine/epinephrine))
+				instantiated_med.metabolizing = FALSE
```

Comment:
this will not work but i think you're aware of that per our discord convo.

Reply:
you should probably look for a trait so we have a source component and we can use it for other instances besides your quirk.

Reply:
yeah i decided to put a reagent_removal_skip_list which is a list of all the causes why this chemical should not be removed from the system. Makes it a lot easier to implement temporary 'stasis' of chemicals inside of somebody, just add your cause to the list, and remove it once you want to resume the normal operation.